File: src/dectobase/dec_to_base.go
package dectobase

import "fmt"

// Iterate :
func Iterate(dec, base int) string {
	var result string
	for dec > 0 {
		result = fmt.Sprintf("%v%s", charset(dec%base), result)
		dec /= base
	}
	return result
}

// Recursive :
func Recursive(dec, base int) string {
	rem := charset(dec % base)
	if dec/base == 0 {
		return rem
	}
	return fmt.Sprintf("%v%v", Recursive(dec/base, base), rem)
}

func charset(num int) string {
	const charset = "0123456789ABCDEF"
	return string(charset[num])
}

File: elixir/lib/dectobase/dec_to_base.ex
defmodule Dectobase.DecToBase do
  def pattern_match(num, base) when div(num, base) == 0 do
    chat_at(rem(num, base))
  end

  def pattern_match(num, base) do
    pattern_match(div(num, base), base) <> chat_at(rem(num, base))
  end

  defp chat_at(pos) do
    String.at("0123456789ABCDEF", pos)
  end

  def std(num, base) do
    Integer.to_string(num, base)
  end
end
